astropy/table/column.py

```diff
@@ -74,26 +74,16 @@ def col_copy(col, copy_indices=True):
     if isinstance(col, BaseColumn):
         return col.copy()
 
-    # The new column should have None for the parent_table ref.  If the
```

Comment:
I think this was a left-over from before `parent_table` was guarded in `data_info` itself.